tests/backend/checkmach/dune

```diff
@@ -1,14 +1,17 @@
+;; CR-someday gyorsh: refactor this file using something like cinaps to
```

Comment:
Cinaps would pull a dependency, but maybe
the technique used in https://github.com/ocaml-flambda/flambda-backend/pull/1244 can be applied
here too.

Reply:
Ah, yes, that's a good idea. I can start by adding a generator for dune these rules, and then simplify the dune file further using the "include files from sexp"  feature when we switch to `dune 3.5`.